analyzer/config/json/json.go

```diff
@@ -36,18 +38,23 @@ func (a ConfigAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput)
 		return nil, xerrors.Errorf("failed to read %s: %w", input.FilePath, err)
 	}
 
-	return &analyzer.AnalysisResult{
-		Files: map[types.HandlerType][]types.File{
-			// It will be passed to misconfig post handler
-			types.MisconfPostHandler: {
-				{
-					Type:    types.JSON,
-					Path:    input.FilePath,
-					Content: b,
+	for _, supportedType := range supportedTypes { // system can contain many json files. Ignore unsupported types.
+		if detection.IsType(input.FilePath, bytes.NewReader(b), supportedType) {
```

Comment:
Looks good, but we can consider how to avoid calling `IsType` twice.
https://github.com/aquasecurity/fanal/blob/03d2949f6ff52825d8fb55bf2bc3795330a0f84b/handler/misconf/misconf.go#L211-L213

I think we can put the detailed type to `Type` instead of `types.JSON`, then skip calling `IsType` here if it is the detailed type.
https://github.com/aquasecurity/fanal/blob/03d2949f6ff52825d8fb55bf2bc3795330a0f84b/handler/misconf/misconf.go#L211-L213